sdk/python/kfp/v2/compiler/pipeline_spec_builder.py

```diff
@@ -376,20 +376,21 @@ def build_container_spec_for_task(
     Returns:
         A PipelineContainerSpec object for the task.
     """
+    if task.container_spec.env is None:
+        task.container_spec.env = {}
+
     container_spec = (
         pipeline_spec_pb2.PipelineDeploymentConfig.PipelineContainerSpec(
             image=task.container_spec.image,
             command=task.container_spec.commands,
             args=task.container_spec.arguments,
+            env=[
+                pipeline_spec_pb2.PipelineDeploymentConfig.PipelineContainerSpec
+                .EnvVar(name=name, value=value)
+                for name, value in task.container_spec.env.items()
```

Comment:
nit: this can be `(task.container_spec.env or {}).items()` so that you don't need the if statement above.